query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverterTest.java

```diff
@@ -1120,14 +1242,19 @@ private QueryRequest buildSimpleQueryWithFilter(Filter filter) {
     return builder.build();
   }
 
-  private void assertPQLQuery(QueryRequest queryRequest, String expectedQuery) {
+  private void assertPQLQuery(
+      QueryRequest queryRequest, String expectedQuery, ViewDefinition viewDefinition) {
+
+    when(mockingExecutionContext.getTenantId()).thenReturn("__default");
```

Comment:
An assertion shouldn't do mocking of its own. I would guess these are not used by most test cases, which means mockito would be erroring out unless we're running in lenient mode (I'm guessing we are). We don't need to switch lenient off as part of this change, but we should be strict for any new tests we write.

https://javadoc.io/static/org.mockito/mockito-core/3.2.4/org/mockito/quality/Strictness.html#STRICT_STUBS

Reply:
`getTenantId()` is used by most of the tests.

Reply:
moved others to the respective test.

Reply:
we shouldn't share the state though - that's the same problem you ran into with view definition. Have each test pass in their own execution context, same as view def, then they can mock that locally, according to their test parameters. For backwards compat, you could overload the assert method with the existing behavior if you want - that way no existing tests hve to be modified. In the current for, we have a mix of both things we don't want - shared state (including newly introduced shared state) and modifying the existing tests.